content/blog/tidymodels-2022-q4/index.Rmd

```diff
@@ -0,0 +1,262 @@
+```
+
+We'll highlight a few especially notable changes below: more specialized role selectors in recipes, extended support for grouped resampling in rsample, and a big speedup in parsnip. First, loading the collection of packages:
+
+```{r}
+#| eval: FALSE
+library(tidymodels)
+```
+
+## Specialized role selectors
+
+The [recipes package for preprocessing](https://recipes.tidymodels.org/) supports tidyselect-style variable selection, and includes some of its own selectors to support common modeling workflows.
+
+To illustrate, we'll make use of a dataset `silly_data` with a number of different variable types:
+
+```{r}
+#| include: FALSE
+assign_letter <- function(names) {
+  c("class", letters[1:(length(names)-1)])
+}
+
+goofy_data <- 
+  sim_classification(num_linear = 3) %>%
+  mutate(across(ends_with("_1"), ~ as.factor(round(.x)))) %>%
+  mutate(two_factor_1 = as.integer(two_factor_1)) %>%
+  rowwise() %>%
+  mutate(char = rnorm(1), char = if_else(char > .5, "maroon", "white")) %>%
+  ungroup() %>%
+  rename_with(assign_letter)
+```
+
+```{r}
+str(goofy_data)
+```
+
+Imagine a classification problem on the `goofy_data` where we'd like to predict `class` using the remaining variables as predictors. The selector functions allow us to perform operations on only the predictors with a certain class. For instance, centering and scaling all numeric predictors:
+
+```{r}
+recipe(class ~ ., goofy_data) %>%
+  step_normalize(all_numeric_predictors()) %>%
+  prep()
+```
+
+Or making dummy variables out of each of the nominal predictors:
+
+```{r}
+recipe(class ~ ., goofy_data) %>%
+  step_dummy(all_nominal_predictors()) %>%
+  prep()
+```
+
+Operations like those above have been long-standing functionality in recipes, and are powerful tools for effective modeling. The most recent release of recipes introduced [finer-grain selectors](https://fosstodon.org/@emilhvitfeldt/109315135944110742) for variable types. For instance, we may want to only center and scale the *double* (i.e. real-valued) predictors, excluding the integers. With the new release of recipes, we can easily do so:
+
+```{r}
+recipe(class ~ ., goofy_data) %>%
+  step_normalize(all_double_predictors()) %>%
+  prep()
+```
+
+This is one of a number of new selectors:
+
+-   The `all_nominal()` selector now has finer-grained versions `all_string()`, `all_factor()`, `all_unordered()`, and `all_ordered()`.
+
+-   The `all_numeric()` selector now has finer-grained versions `all_double()`, and `all_integer()`.
+
+-   New `all_logical()`, `all_date()`, and `all_datetime()` selectors.
+
+All new selectors have `*_predictors()` variants. You can read more about recipes 1.0.3 in the [release notes](https://recipes.tidymodels.org/news/index.html#recipes-103).
+
+## Grouped resampling
+
+The most recent release of rsample introduced support for stratification with grouped resampling. Consider the following data set, giving whether a number of `household`s `chops` the `n_melons` melons in their fridge:
```

Comment:
I think it's a little easier to explain the variables after the code chunk (avoiding a little grammar clash between households plural and chops singular):

```suggestion
The most recent release of rsample introduced support for stratification with grouped resampling. Consider the following toy data set on the number of melons in a household:
```

Reply:
Agreed.